sktime/forecasting/var.py

```diff
@@ -184,6 +185,112 @@ def _predict(self, fh, X=None):
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon
+            The forecasting horizon with the steps ahead to to predict.
+        X : optional (default=None)
+            guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+        coverage : list of float (guaranteed not None and floats in [0,1] interval)
+           nominal coverage(s) of predictive interval(s)
+
+        Returns
+        -------
+        pred_int : pd.DataFrame
+            Column has multi-index: first level is variable name from y in fit,
+                second level coverage fractions for which intervals were computed.
+                    in the same order as in input `coverage`.
+                Third level is string "lower" or "upper", for lower/upper interval end.
+            Row index is fh, with additional (upper) levels equal to instance levels,
+                from y seen in fit, if y_inner_mtype is Panel or Hierarchical.
+            Entries are forecasts of lower/upper interval end,
+                for var in col index, at nominal coverage in second col index,
+                lower/upper depending on third col index, for the row index.
+                Upper/lower interval end forecasts are equivalent to
+                quantile forecasts at alpha = 0.5 - c/2, 0.5 + c/2 for c in coverage.
+        """
+        model = self._fitted_forecaster
+        fh_int = fh.to_relative(self.cutoff)
+        steps = fh_int[-1]
+        n_lags = model.k_ar
+
+        y_cols_no_space = [str(col).replace(" ", "") for col in self._y.columns]
+
+        df_list = []
+
+        for cov in coverage:
+
+            alpha = -0.5 * cov + 0.5
+
+            fcast_interval = model.forecast_interval(
+                self._y.values[-n_lags:], steps=steps, alpha=alpha
+            )
+            lower_int, upper_int = fcast_interval[1], fcast_interval[-1]
+
+            lower_df = pd.DataFrame(
+                lower_int,
+                columns=[
+                    col + " " + str(alpha) + " " + "lower" for col in y_cols_no_space
+                ],
+            )
+            upper_df = pd.DataFrame(
+                upper_int,
+                columns=[
+                    col + " " + str(alpha) + " " + "upper" for col in y_cols_no_space
+                ],
+            )
+
+            df_list.append(pd.concat((lower_df, upper_df), axis=1))
+
+        concat_df = pd.concat(df_list, axis=1)
+
+        concat_df_columns = {
+            col_df
+            for col in y_cols_no_space
+            for col_df in concat_df.columns
+            if col in col_df
+        }
+
+        pre_output_df = concat_df[list(concat_df_columns)]
+
+        pre_output_df_2 = pd.DataFrame(
+            pre_output_df.values,
+            columns=pd.MultiIndex.from_tuples(
+                [col.split(" ") for col in pre_output_df]
+            ),
+        )
+
+        final_columns = [
```

Comment:
ising `zip` might be more elegant than a loop?

Reply:
I am not sure how you suggest using `zip`, wouldn't `zip` just compute the internal product of the three lists (?) when we actually want the Cartesian product. I did something neater (but more complicated and needs a package import) in the upcoming PR.

Reply:
ah, you are right, sorry. You want `itertools.product`, not `zip`.

Reply:
... which you actually used, all fine then!